### ISSUE TITLE
toHaveBeenCalledWith does not fail gracefully when called on a non-spy

### ISSUE THREAD

**Issue by Jameskmonger** [alsatian-test/alsatian#153]
```
interface IDependency {
    someVoid (arg1: string): void;
}

class SomeClass {
    constructor (dependency: IDependency) {

    }

    someAction (): void {

    }
}

export class SomeTest {
    @Test()
    public shouldCallSomeVoidOnDependency(): void {
        let dep: IDependency = {
            someVoid: (arg1: string) => { }
        };

        SpyOn(dep, "someVoid");

        let classInstance = new SomeClass(dep);

        Expect(dep).toHaveBeenCalledWith("bob");
    }
}
```

This gives the following output which isn't very user friendly:
```
# ERROR: Cannot read property 'filter' of undefined
```

This is thrown by `toHaveBeenCalledWith` in `expect.ts`.

I think it would be better to explicitly state that it's not a spy.

**Comment by jamesrichford** [comment]
To be fixed as part of #155 which should also check types and throw appropriate errors

**Comment by jamesrichford** [comment]
closed by #167
